### PR TITLE
Refactor of methods used to display messages [PD-1050]

### PR DESCRIPTION
**Pull request by tdphillips**
This was not fun.

I decided to see if I could get by with a context processor and it appears to work. If it is decided that this is not the way to go, I am at the point where trashing this branch would almost be preferable.

All tests pass. Let's see if Jenkins agrees.

### CODE REVIEW

**Comment by tdphillips** [comment]
Before this gets merged, if it does, duplicate messages need to be cleared.

File: mymessages/tests/test_models.py [hunk 1]
```diff
@@ -28,12 +28,10 @@ def test_message_made(self):
         self.assertGreaterEqual(m, 1)
 
     def test_message_made_sent_to_multiple(self):
```
Comment (aspidites) [inline mymessages/tests/test_models.py]: Can we give these variables meaningful names and document the test?

**Comment by aspidites** [comment]
From what I saw, it looks much better. Will take it out for a test spin once my database isn't in a state of flux.

**Comment by tdphillips** [comment]
I still duplicate a little bit as far as the inbox is concerned. I'm not sure I would be happy with any solution at this point.

**Comment by aspidites** [comment]
We probably want a data migration to take care of this:
```bash
Running migrations for mymessages:
 - Migrating forwards to 0003_auto__add_unique_messageinfo_user_message.
 > mymessages:0003_auto__add_unique_messageinfo_user_message
FATAL ERROR - The following SQL query failed: ALTER TABLE `mymessages_messageinfo` ADD CONSTRAINT `mymessages_messageinfo_user_id_6815c0f0d728d9b3_uniq` UNIQUE (`user_id`, `message_id`)
The error was: (1062, "Duplicate entry '58472-11' for key 'mymessages_messageinfo_user_id_6815c0f0d728d9b3_uniq'")
 ! Error found during real run of migration! Aborting.

 ! Since you have a database that does not support running
 ! schema-altering statements in transactions, we have had 
 ! to leave it in an interim state between migrations.

! You *might* be able to recover with:   - no dry run output for delete_unique_column() due to dynamic DDL, sorry

 ! The South developers regret this has happened, and would
 ! like to gently persuade you to consider a slightly
 ! easier-to-deal-with DBMS (one that supports DDL transactions)
 ! NOTE: The error which caused the migration to fail is further up.
Error in migration: mymessages:0003_auto__add_unique_messageinfo_user_message
Traceback (most recent call last):
  File "manage.py", line 9, in <module>
    execute_from_command_line(sys.argv)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/management/commands/migrate.py", line 111, in handle
    ignore_ghosts = ignore_ghosts,
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/__init__.py", line 220, in migrate_app
    success = migrator.migrate_many(target, workplan, database)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/migrators.py", line 254, in migrate_many
    result = migrator.__class__.migrate_many(migrator, target, migrations, database)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/migrators.py", line 329, in migrate_many
    result = self.migrate(migration, database)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/migrators.py", line 133, in migrate
    result = self.run(migration, database)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/migrators.py", line 114, in run
    return self.run_migration(migration, database)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/migrators.py", line 84, in run_migration
    migration_function()
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/migration/migrators.py", line 60, in <lambda>
    return (lambda: direction(orm))
  File "/home/edwin/projects/myjobs/mymessages/migrations/0003_auto__add_unique_messageinfo_user_message.py", line 12, in forwards
    db.create_unique(u'mymessages_messageinfo', ['user_id', 'message_id'])
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/db/generic.py", line 47, in _cache_clear
    return func(self, table, *args, **opts)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/db/generic.py", line 623, in create_unique
    cols,
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/south/db/generic.py", line 282, in execute
    cursor.execute(sql, params)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/db/backends/util.py", line 69, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/django/db/backends/mysql/base.py", line 124, in execute
    return self.cursor.execute(query, args)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/MySQLdb/cursors.py", line 174, in execute
    self.errorhandler(self, exc, value)
  File "/home/edwin/.virtualenvs/myjobs/lib/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
django.db.utils.IntegrityError: (1062, "Duplicate entry '58472-11' for key 'mymessages_messageinfo_user_id_6815c0f0d728d9b3_uniq'")

```

**Comment by tdphillips** [comment]
That means we should probably have a discussion on which one to keep.

**Comment by aspidites** [comment]
> That means we should probably have a discussion on which one to keep.
@JLMcLaugh ?

**Comment by tdphillips** [comment]
Added a data migration. Due to the way I constructed it, it keeps the most recent MessageInfo for a given (user, message) combo.

**Pull request merged**